roles/edpm_network_config/tasks/os_net_config.yml

```diff
@@ -30,6 +30,16 @@
         dest: "{{ nic_config_file }}"
         mode: '0644'
         backup: true
+    - name: Retrieve and output nic_config_file contents for debug before applying
+      when: edpm_network_config_debug|bool
+      block:
+        - name: Retrieve content of nic_config_file before applying
+          ansible.builtin.slurp:
+            path: "{{ nic_config_file }}"
+          register: os_net_config_config
+        - name: Debug print nic_config_file contents
+          ansible.builtin.debug:
+            msg: "{{ os_net_config_config | b64decode | trim }}"
```

Comment:
`os_net_config_config` is a dict not the b64 content. To decode that content you need:
`os_net_config_config['content'] | b64decode`

Reply:
yes right updating sorry did that too quickly :/

Reply:
(actually during the reviews call ;))